### PR TITLE
PUBDEV-8094: Restructuring algorithm parameters and grid search hyperparameters

### PR DESCRIPTION
**Pull request by hannah-tillman**
This is the draft PR for the new restructuring of the algorithm parameters.

What I need in the initial review:

- [ ] are the algorithm-specific parameters correct for each algorithm
- [ ] does it make sense to have the sub-groups for:
        - [ ] GLM Family parameters (i.e., shared parameters for GLM, GAM, MS, & ANOVA)
        - [ ] Tree algorithm parameters (i.e., shared parameters for GBM, DRF, XGBoost, Isolation Forest, Ex. Isolation Forest)
- [ ] are the hyperparameter lists on the grid search page correct
        - [ ] does it make sense to have the shared tables for:
                 - [ ] common hyperparameters for supervised algos
                 - [ ] common hyperparameters for unsupervised algos
                 - [ ] shared tree algorithm hyperparameters
                 - [ ] shared GLM family hyperparameters

Let me know if you need help building the docs to look through them easier or if anything doesn't make sense!

### CODE REVIEW

File: h2o-docs/src/product/data-science/anova_glm.rst [hunk 1]
```diff
@@ -11,134 +11,172 @@ Since ANOVA GLM is mainly used to investigate the contribution of each predictor
...
--  `weights_column <algo-params/weights_column.html>`__: Specify a column to use for the observation
-   weights, which are used for bias correction. The specified
-   ``weights_column`` must be included in the specified
-   ``training_frame``. 
-   
-    *Python only*: To use a weights column when passing an H2OFrame to ``x`` instead of a list of column names, the specified ``training_frame`` must contain the specified ``weights_column``. 
-   
-    **Note**: Weights are per-row observation weights and do not increase the size of the data frame. This is typically the number of times a row is repeated, but non-integer values are supported as well. During training, rows with higher weights matter more, due to the larger loss function pre-factor.
+Maybe algorithm-specific
+''''''''''''''''''''''''
```
Comment (wendycwong) [inline h2o-docs/src/product/data-science/anova_glm.rst:37]: The following parameters are specific to GLM, GAM, Modelselection, ANOVAGLM.  So not sure how you want to call them.

**Review by wendycwong — CHANGES_REQUESTED** [review]

File: h2o-docs/src/product/data-science/gam.rst [hunk 2]
```diff
@@ -16,213 +16,260 @@ GAM supports importing and exporting `MOJOs <../save-and-load-model.html#support
...
+            ‘gam_columns’:[[“C2",“C3”]]
+        }]
+   }
 
--  `ignore_const_cols <algo-params/ignore_const_cols.html>`__: Enable this option to ignore constant
-   training columns, since no information can be gained from them. This option is defaults to true (enabled).
+With this hyperparameter specification, there will be 16 GAM models built.
 
--  `score_each_iteration <algo-params/score_each_iteration.html>`__: (Optional) Enable this option to score during each iteration of the model training. This option is defaults to false (not enabled).
+GLM family parameters
```
[the remark this answers is not in the document]
Comment (hannah-tillman) [reply h2o-docs/src/product/data-science/gam.rst:72]: @wendycwong Thoughts on something like this? Having a link on GAM, MS, and ANOVA linking to a shared parameter section on the GLM page where each section is laid out?

https://user-images.githubusercontent.com/52463461/202790160-dfeeb738-58a5-421c-bdc0-f95ee33dd656.mov

[the remark this answers is not in the document]
Comment (hannah-tillman) [reply h2o-docs/src/product/data-science/gam.rst:72]: Or maybe something more like this with small icons for which algos share the parameter:
Screen Shot 2022-11-18 at 2 47 39 PM

**Comment by haohenggang** [comment]
@wendycwong @vindows @honzasterba @pkozelka

**Comment by haohenggang** [comment]
@wendycwong @vindows @honzasterba @pkozelka Could you please share me if there is  any examples of using random forests algorithm by java or Android environment？

File: h2o-docs/src/product/data-science/gam.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (wendycwong) [reply h2o-docs/src/product/data-science/gam.rst:48]: Yes, this looks very nice indeed!

**Comment by wendycwong** [comment]
@haohenggang Here is a code snippet on how to use random forest:

            Frame fr = Scope.track(parseTestFile("smalldata/junit/titanic_alt.csv"));
            DRFModel.DRFParameters parms = new DRFModel.DRFParameters();
            parms._train = fr._key;
            parms._distribution = gaussian;
            parms._response_column = "age";
            parms._ntrees = 5;
            parms._max_depth = 4;
            parms._min_rows = 1;
            parms._nbins = 50;
            parms._score_each_iteration = true;
            parms._seed = 42;

            DRF job = new DRF(parms);
            DRFModel drf = job.trainModel().get();

**Comment by hannah-tillman** [comment]
Resolved in subtasks PUBDEV-9051 through PUBDEV-9071. Part of 3.40.0.4 release.

**Pull request closed without merging**